ts/api/api.ts

```diff
@@ -123,6 +124,11 @@ export interface OpenYoloApi {
    * Cancels the last pending OpenYOLO request.
    */
   cancelLastOperation(): Promise<void>;
+
+  /**
+   * Sets the desired log level for the OpenYOLO library.
+   */
+  setLogLevel(level: LogLevel): Promise<void>;
```

Comment:
Agreed it's not a great name... I got confused at first too. I am not sure overriding per request is useful. Developer will usually set the level to `DEBUG` in their dev environment and that's it, I think.

Also, the way you're currently doing will not allow providers to also use the logging level for logging their internal implementation, as it will never be exposed to them. I'm in favor of moving that to the `OnDemandOpenYoloApi` interface and use the query parameter method, but feel free to also ask Iain for his opinion.